packages/web-components/fast-foundation/src/select/select.ts

```diff
@@ -244,9 +244,7 @@ export class Select extends FormAssociatedSelect {
      */
     public formResetCallback(): void {
         this.setProxyOptions();
-        // Call the base class's implementation setDefaultSelectedOption instead of the select's
-        // override, in order to reset the selectedIndex without using the value property.
-        super.setDefaultSelectedOption();
+        this.selectedIndex = 0;
```

Comment:
I think calling the `super.setDefaultSelectedOption()` can still happen, but there should be a check here that if `selectedIndex` is -1, then it should be set to 0:

```ts
    public formResetCallback(): void {
        this.setProxyOptions();
        super.setDefaultSelectedOption();
        if (this.selectedIndex === -1) {
            this.selectedIndex = 0;
        }
    }
```

(This check will come in handy in #5595, where we'll need to modify it to check when `this.collapsible` is true)

Reply:
I've gone ahead and made the change that I suggested in case you agree that that's a sufficient solution.

Reply:
The [`formRestCallback()`](https://web.dev/more-capable-form-controls/#void-formresetcallback()) function only affects `<fast-select>` when its associated form is reset. This function isn't present nor does it affect the `<fast-listbox>` component, since that component isn't form-associated. Because the `<fast-select>` can't currently be expressed with the `size` or `multiple` attributes, we can't reset the `selectedIndex` to -1 since [this wouldn't match the expected behavior of the native select](https://html.spec.whatwg.org/multipage/form-elements.html#:~:text=If%20the%20select%20element%27s%20display%20size%20is%201%2C%20and%20no%20option%20elements%20in%20the%20select%20element%27s%20list%20of%20options%20have%20their%20selectedness%20set%20to%20true). We also can't reset it to 0 without checking the default selectedness of the options.

By calling `super.setDefaultSelectedOptions()` instead of `this.setDefaultSelectedOptions()` we avoid re-calling the Select's method override, which has to handle the situation where the component is connected before its child options are connected. This is unnecessary in `formResetCallback()`, since by the time a form is reset, it's expected that the component and all of its child options are already connected and present in the DOM.

Reply:
As for the comment in the code: there's a somewhat back-and-forth behavior between the `value` property and the `selectedIndex` property. Programmatically setting the select's `value` property directly will trigger a search for a matching option, then set the `selectedIndex` to match that option's index if one is found. Likewise, by programmatically setting `selectedIndex`, it has to set the select's `value` property to match the value of the option at that index. This is made slightly weirder since the `<fast-select>` component doesn't have a `value` _attribute_; its value is always determined by the selectedness of its options. The select's value can only be set to match one of its options, and setting it to anything without a match resets the value to nothing and sets the `selectedIndex` to -1.

Reply:
Ah, my mistake. I jumped to a false conclusion after I re-ran the test suite. Your suggestion works just fine.